src/injected/visualization-state-change-handler.ts

```diff
@@ -1,35 +1,30 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
-import { VisualizationType } from 'common/types/visualization-type';
+import { VisualizationConfigurationFactory } from 'common/configs/visualization-configuration-factory';
 
-import { AssessmentsProvider } from '../assessments/types/assessments-provider';
 import { TargetPageStoreData } from './client-store-listener';
 import { UpdateVisualization } from './target-page-visualization-updater';
 
 export class VisualizationStateChangeHandler {
     constructor(
-        private visualizations: VisualizationType[],
         private visualizationUpdater: UpdateVisualization,
-        private assessmentProvider: AssessmentsProvider,
+        private visualizationConfigurationFactory: VisualizationConfigurationFactory,
     ) {}
 
     public updateVisualizationsWithStoreData = async (storeData: TargetPageStoreData) => {
         if (!storeData.assessmentStoreData) {
             return;
         }
-        await Promise.all(
-            this.visualizations.map(async visualizationType => {
-                if (this.assessmentProvider.isValidType(visualizationType)) {
-                    const stepMap = this.assessmentProvider.getStepMap(visualizationType);
-                    await Promise.all(
-                        Object.values(stepMap).map(step =>
-                            this.visualizationUpdater(visualizationType, step.key, storeData),
-                        ),
-                    );
-                } else {
-                    await this.visualizationUpdater(visualizationType, null, storeData);
-                }
-            }),
+
+        const updateCalls = [];
+        this.visualizationConfigurationFactory.forEachConfig(
+            (testConfig, type, requirementConfig) => {
```

Comment:
```suggestion
            (testConfig: VisualizationConfiguration, type: VisualizationType, requirementConfig: Requirement) => {
```

Explicit types increase readability.